### PR TITLE
Topic 01 js dom ap is

### PR DESCRIPTION
**Pull request by NicolasDF**
Topic 01 Done.

### CODE REVIEW

File: 01-JS-DOM-APIs/index.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <head>
+        <meta charset="UTF-8"> 
+        <meta name="description" content="Globant Bootcamp Javascript Topic">
+        <meta name="keywords" content="Bootcamp,practise,javascript">
+        <meta name="author" content="Nicolas Del Fiorentino">
+        <meta name="viewport" content="width=device-width,initial-scale=1"> 
+
+        <link href="styles/normalize.css" rel="stylesheet">
+        <link href="styles/index.css" rel="stylesheet">
+
```
Comment (elnawe) [inline 01-JS-DOM-APIs/index.html]: Lots of break lines here.

File: 01-JS-DOM-APIs/index.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <meta name="viewport" content="width=device-width,initial-scale=1"> 
+
+        <link href="styles/normalize.css" rel="stylesheet">
+        <link href="styles/index.css" rel="stylesheet">
+        
+
+        <title>Topic 1</title>
+    </head>
+
+	<body>
```
Comment (elnawe) [inline 01-JS-DOM-APIs/index.html]: Fix indentation in this file.

File: 01-JS-DOM-APIs/index.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		<header>
+			<h1 class="title">Super JavaScript webpage</h1>
+		</header>
+		<section id="hide" class="section">
+			Hello world!
+		</section>  
+        <div>
+            <button id="btn_joke" class="button" type="button">Get a Joke</button>
```
Comment (elnawe) [inline 01-JS-DOM-APIs/index.html]: do not use short naming.

File: 01-JS-DOM-APIs/script.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+function showHidden () {	
+	var sec = document.getElementById('hide');
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: fix indentation. break line after variable declaration.

File: 01-JS-DOM-APIs/script.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    if (sec.style.display =='block')
+    {
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: this curly should be just after conditional

File: 01-JS-DOM-APIs/script.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        sec.style.display=='none';
+    } else {
+        var timer = setInterval(function(){
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: space just before parenthesis.

File: 01-JS-DOM-APIs/script.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            sec.style.display = 'block';
+            sec.className = 'section_fadeIn';
+        }, 1000)
+    }
+}
+
+function getJoke () {
+	var request = new XMLHttpRequest();
+	request.open('GET', 'http://api.icndb.com/jokes/random', true);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: break line before this one.

File: 01-JS-DOM-APIs/script.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+	request.onload = function() {
+	  if (request.status >= 200 && request.status < 400) {
+	    var data = JSON.parse(request.response);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: fix indentation here. Review this file.

File: 01-JS-DOM-APIs/script2.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+function createDom () {
+	var structure = [
+		['Persona', 'Edad'],
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: You should use english here.

File: 01-JS-DOM-APIs/script2.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+		['Nicolas', 22],
+		['Carlos', 45],
+		['Martiano', 35],
+		['Jose', 64]
+	];
+	var newTable = document.createElement('table');
+	newTable.id = 'table';
+	newTable.className = 'table';
+	document.getElementById('target').appendChild(newTable);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: Break line before and after this.

File: 01-JS-DOM-APIs/script2.js [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	var newTr =  document.createElement('tr');
+	newTr.className = 'tr';		
+	document.getElementById('table').appendChild(newTr);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: Break line before this one.

File: 01-JS-DOM-APIs/script2.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+		var newTh = document.createElement('th');
+		newTh.className = 'th';
+		newTh.innerHTML = structure[0][i];
+		document.getElementsByClassName('tr')[0].appendChild(newTh);
+	};
+		
+	for (let i = 1; i < structure.length; i++) {
+		var newTr =  document.createElement('tr');
+		newTr.className = 'tr';		
+		document.getElementById('table').appendChild(newTr);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: Break line before and after this one.

File: 01-JS-DOM-APIs/script2.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+	document.getElementById('target').appendChild(newTable);
+	var newTr =  document.createElement('tr');
+	newTr.className = 'tr';		
+	document.getElementById('table').appendChild(newTr);
+
+	for (let i = 0 ; i < structure[0].length; i++) {		
+		var newTh = document.createElement('th');
+		newTh.className = 'th';
+		newTh.innerHTML = structure[0][i];
+		document.getElementsByClassName('tr')[0].appendChild(newTh);
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: Break line before this.

File: 01-JS-DOM-APIs/secondpage.html [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+        <meta name="keywords" content="Bootcamp,practise,javascript">
+        <meta name="author" content="Nicolas Del Fiorentino">
+        <meta name="viewport" content="width=device-width,initial-scale=1"> 
+
+        <link href="styles/normalize.css" rel="stylesheet">
+        <link href="styles/index.css" rel="stylesheet">
+
+        <title>Topic 1 - Exercise 12</title>
+    </head>
+	<body id="target">
```
Comment (elnawe) [inline 01-JS-DOM-APIs/secondpage.html]: Fix indentation here.

File: 01-JS-DOM-APIs/styles/index.css [hunk 15]
```diff
@@ -0,0 +1,42 @@
+.button {
+  font-size: 20px;
```
Comment (elnawe) [inline 01-JS-DOM-APIs/styles/index.css]: Review indentation. Should be 4 spaces.

File: 01-JS-DOM-APIs/styles/index.css [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+.title {
+	text-align: center;
+}
+
+.table, .tr, .td {
+  border-collapse: collapse;
+  border-spacing: 0;
+  empty-cells: show;
+  border: 1px solid #cbcbcb;
```
Comment (elnawe) [inline 01-JS-DOM-APIs/styles/index.css]: Alpha order! (review whole file)

File: 01-JS-DOM-APIs/script.js [hunk 17]
```diff
@@ -1,10 +1,11 @@
 function showHidden () {	
 	var sec = document.getElementById('hide');
+
     if (sec.style.display =='block')
     {
         sec.style.display=='none';
     } else {
-        var timer = setInterval(function(){
+        var timer = setInterval (function(){
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: This is not the space I was talking about, I meant the other one, after `function` :P

File: 01-JS-DOM-APIs/script.js [hunk 18]
```diff
@@ -1,10 +1,11 @@
 function showHidden () {	
 	var sec = document.getElementById('hide');
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: It looks like this line has more indentation that the others

File: 01-JS-DOM-APIs/script.js [hunk 19]
```diff
@@ -13,18 +14,19 @@ function showHidden () {
...
-	  if (request.status >= 200 && request.status < 400) {
-	    var data = JSON.parse(request.response);	    
-	    var newP = document.createElement('p');
-	    newP.className = 'joke';
-	    newP.innerHTML = data.value.joke;
-	    document.getElementById('section_joke').appendChild(newP);	    
-	  } else {
-	    console.log(Error(reques.statusText));
-	  }
+		if (request.status >= 200 && request.status < 400) {
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script.js]: You added a tab here instead of spaces?

File: 01-JS-DOM-APIs/script2.js [hunk 20]
```diff
@@ -1,32 +1,39 @@
...
 	var structure = [
-		['Persona', 'Edad'],
+		['Name', 'Age'],
 		['Nicolas', 22],
 		['Carlos', 45],
 		['Martiano', 35],
 		['Jose', 64]
 	];
 	var newTable = document.createElement('table');
+
```
Comment (elnawe) [inline 01-JS-DOM-APIs/script2.js]: Remember:

Separate with a break line

- declarations from methods
- return
- if statements.

File: 01-JS-DOM-APIs/secondpage.html [hunk 21]
```diff
@@ -9,12 +9,10 @@
...
         <link href="styles/normalize.css" rel="stylesheet">
         <link href="styles/index.css" rel="stylesheet">
-
         <title>Topic 1 - Exercise 12</title>
     </head>
 	<body id="target">
-        <h1>Topic 1 - Exercise 12</h1>
-
+        <h1>Topic 1 - Exercise 12</h1>        
     <script src="script2.js"></script>
```
Comment (elnawe) [inline 01-JS-DOM-APIs/secondpage.html]: This need more indentation.

File: 01-JS-DOM-APIs/script2.js (hunk 20 above)
Comment (NicolasDF) [inline 01-JS-DOM-APIs/script2.js]: After the array, should i leave a break line?

File: 01-JS-DOM-APIs/script.js (hunk 17 above)
Comment (NicolasDF) [inline 01-JS-DOM-APIs/script.js]: it should be var timer = setInterval( function () {   
Right?

**Pull request merged**